### ISSUE TITLE
java.lang.IllegalStateException: Unable to comment the commit

### ISSUE THREAD

**Issue by jangrewe** [gabrie-allaigre/sonar-gitlab-plugin#95]
Hi, my GitLab instance throws a 500 when your plugin tries to add a comment:
```
ActiveRecord::StatementInvalid (Mysql2::Error: Incorrect string value: '\xF0\x9F\x9A\xAB</...' for column 'note_html' at row 1: INSERT INTO `notes` (`note`, `noteable_type`, `commit_id`, `project_id`, `author_id`, `discussion_id`, `created_at`, `updated_at`, `note_html`, `cached_markdown_version`, `system`) VALUES ('SonarQube analysis reported 6 issues\n* :no_entry_sign: 1 critical\n* :warning: 3 major\n* :arrow_down_small: 1 minor\n* :information_source: 1 info\n\nNote: The following issues were found on lines that were not modified in the commit. Because these issues can\'t be reported as line comments, they are summarized here:\n\n1. :no_entry_sign: [Refactor this method to reduce its Cognitive Complexity from 22 to the 15 allowed.](https://git.faked.org/jan/org.faked.isms2droid/blob/1b9fcb99d18685340f21db583ce4ac3eec0f0c0f/app/src/main/java/org/faked/isms2droid/CallsActivity.java#L284) [:blue_book:](https://sonar.faked.org/coding_rules#rule_key=squid%3AS3776)\n1. :warning: [Rename this method or correct the type of the argument(s) to override the parent class method.](https://git.faked.org/jan/org.faked.isms2droid/blob/1b9fcb99d18685340f21db583ce4ac3eec0f0c0f/app/src/main/java/org/faked/isms2droid/CallsActivity.java#L180) [:blue_book:](https://sonar.faked.org/coding_rules#rule_key=squid%3AS2177)\n1. :warning: [Rename this method or correct the type of the argument(s) to override the parent class method.](https://git.faked.org/jan/org.faked.isms2droid/blob/1b9fcb99d18685340f21db583ce4ac3eec0f0c0f/app/src/main/java/org/faked/isms2droid/CallsActivity.java#L195) [:blue_book:](https://sonar.faked.org/coding_rules#rule_key=squid%3AS2177)\n1. :warning: [Remove this unused method parameter \"dummy\".](https://git.faked.org/jan/org.faked.isms2droid/blob/1b9fcb99d18685340f21db583ce4ac3eec0f0c0f/app/src/main/java/org/faked/isms2droid/CallsActivity.java#L195) [:blue_book:](https://sonar.faked.org/coding_rules#rule_key=squid%3AS1172)\n1. :arrow_down_small: [Remove the declaration of thrown exception \'java.io.IOException\', as it cannot be thrown from method\'s body.](https://git.faked.org/jan/org.faked.isms2droid/blob/1b9fcb99d18685340f21db583ce4ac3eec0f0c0f/app/src/main/java/org/faked/isms2droid/CallsActivity.java#L226) [:blue_book:](https://sonar.faked.org/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)\n1. :information_source: [Complete the task associated to this TODO comment.](https://git.faked.org/jan/org.faked.isms2droid/blob/1b9fcb99d18685340f21db583ce4ac3eec0f0c0f/app/src/main/java/org/faked/isms2droid/CallsActivity.java#L118) [:blue_book:](https://sonar.faked.org/coding_rules#rule_key=squid%3AS1135)\n', 'Commit', '1b9fcb99d18685340f21db583ce4ac3eec0f0c0f', 12, 2, 'f36a2ebc8df8817a22f7736038177c7461652709', '2018-01-21 09:59:37', '2018-01-21 09:59:37', '<p dir=\"auto\">SonarQube analysis reported 6 issues</p>\n\n<ul dir=\"auto\">\n<li>\n<gl-emoji title=\"no entry sign\" data-name=\"no_entry_sign\" data-unicode-version=\"6.0\">🚫</gl-emoji> 1 critical</li>\n<li>\n<gl-emoji title=\"warning sign\" data-name=\"warning\" data-unicode-version=\"4.0\">⚠</gl-emoji> 3 major</li>\n<li>\n<gl-emoji title=\"down-pointing small red triangle\" data-name=\"arrow_down_small\" data-unicode-version=\"6.0\">🔽</gl-emoji> 1 minor</li>\n<li>\n<gl-emoji title=\"information source\" data-name=\"information_source\" data-unicode-version=\"3.0\">ℹ</gl-emoji> 1 info</li>\n</ul>\n\n<p dir=\"auto\">Note: The following issues were found on lines that were not modified in the commit. Because these issues can\'t be reported as line comments, they are summarized here:</p>\n\n<ol dir=\"auto\">\n<li>\n<gl-emoji title=\"no entry sign\" data-name=\"no_entry_sign\" data-unicode-version=\"6.0\">🚫</gl-emoji> <a href=\"https://git.faked.org/jan/org.faked.isms2droid/blob/1b9fcb99d18685340f21db583ce4ac3eec0f0c0f/app/src/main/java/org/faked/isms2droid/CallsActivity.java#L284\">Refactor this method to reduce its Cognitive Complexity from 22 to the 15 allowed.</a> <a href=\"https://sonar.faked.org/coding_rules#rule_key=squid%3AS3776\" rel=\"nofollow noreferrer noopener\" target=\"_blank\"><gl-emoji title=\"blue book\" data-name=\"blue_book\" data-unicode-version=\"6.0\">📘</gl-emoji></a>\n</li>\n<li>\n<gl-emoji title=\"warning sign\" data-name=\"warning\" data-unicode-version=\"4.0\">⚠</gl-emoji> <a href=\"https://git.faked.org/jan/org.faked.isms2droid/blob/1b9fcb99d18685340f21db583ce4ac3eec0f0c0f/app/src/main/java/org/faked/isms2droid/CallsActivity.java#L180\">Rename this method or correct the type of the argument(s) to override the parent class method.</a> <a href=\"https://sonar.faked.org/coding_rules#rule_key=squid%3AS2177\" rel=\"nofollow noreferrer noopener\" target=\"_blank\"><gl-emoji title=\"blue book\" data-name=\"blue_book\" data-unicode-version=\"6.0\">📘</gl-emoji></a>\n</li>\n<li>\n<gl-emoji title=\"warning sign\" data-name=\"warning\" data-unicode-version=\"4.0\">⚠</gl-emoji> <a href=\"https://git.faked.org/jan/org.faked.isms2droid/blob/1b9fcb99d18685340f21db583ce4ac3eec0f0c0f/app/src/main/java/org/faked/isms2droid/CallsActivity.java#L195\">Rename this method or correct the type of the argument(s) to override the parent class method.</a> <a href=\"https://sonar.faked.org/coding_rules#rule_key=squid%3AS2177\" rel=\"nofollow noreferrer noopener\" target=\"_blank\"><gl-emoji title=\"blue book\" data-name=\"blue_book\" data-unicode-version=\"6.0\">📘</gl-emoji></a>\n</li>\n<li>\n<gl-emoji title=\"warning sign\" data-name=\"warning\" data-unicode-version=\"4.0\">⚠</gl-emoji> <a href=\"https://git.faked.org/jan/org.faked.isms2droid/blob/1b9fcb99d18685340f21db583ce4ac3eec0f0c0f/app/src/main/java/org/faked/isms2droid/CallsActivity.java#L195\">Remove this unused method parameter \"dummy\".</a> <a href=\"https://sonar.faked.org/coding_rules#rule_key=squid%3AS1172\" rel=\"nofollow noreferrer noopener\" target=\"_blank\"><gl-emoji title=\"blue book\" data-name=\"blue_book\" data-unicode-version=\"6.0\">📘</gl-emoji></a>\n</li>\n<li>\n<gl-emoji title=\"down-pointing small red triangle\" data-name=\"arrow_down_small\" data-unicode-version=\"6.0\">🔽</gl-emoji> <a href=\"https://git.faked.org/jan/org.faked.isms2droid/blob/1b9fcb99d18685340f21db583ce4ac3eec0f0c0f/app/src/main/java/org/faked/isms2droid/CallsActivity.java#L226\">Remove the declaration of thrown exception \'java.io.IOException\', as it cannot be thrown from method\'s body.</a> <a href=\"https://sonar.faked.org/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck\" rel=\"nofollow noreferrer noopener\" target=\"_blank\"><gl-emoji title=\"blue book\" data-name=\"blue_book\" data-unicode-version=\"6.0\">📘</gl-emoji></a>\n</li>\n<li>\n<gl-emoji title=\"information source\" data-name=\"information_source\" data-unicode-version=\"3.0\">ℹ</gl-emoji> <a href=\"https://git.faked.org/jan/org.faked.isms2droid/blob/1b9fcb99d18685340f21db583ce4ac3eec0f0c0f/app/src/main/java/org/faked/isms2droid/CallsActivity.java#L118\">Complete the task associated to this TODO comment.</a> <a href=\"https://sonar.faked.org/coding_rules#rule_key=squid%3AS1135\" rel=\"nofollow noreferrer noopener\" target=\"_blank\"><gl-emoji title=\"blue book\" data-name=\"blue_book\" data-unicode-version=\"6.0\">📘</gl-emoji></a>\n</li>\n</ol>', 2, 0)):
```

The output of the plugin:
```
SonarQube failed to complete the review of this commit
java.lang.IllegalStateException: Unable to comment the commit
	at com.talanlabs.sonar.plugins.gitlab.GitLabApiV3Wrapper.addGlobalComment(GitLabApiV3Wrapper.java:280)
	at com.talanlabs.sonar.plugins.gitlab.CommitFacade.addGlobalComment(CommitFacade.java:136)
	at com.talanlabs.sonar.plugins.gitlab.CommitIssuePostJob.updateGlobalComments(CommitIssuePostJob.java:190)
	at com.talanlabs.sonar.plugins.gitlab.CommitIssuePostJob.execute(CommitIssuePostJob.java:75)
	at org.sonar.scanner.postjob.PostJobWrapper.executeOn(PostJobWrapper.java:58)
	at org.sonar.scanner.phases.PostJobsExecutor.execute(PostJobsExecutor.java:66)
	at org.sonar.scanner.phases.PostJobsExecutor.execute(PostJobsExecutor.java:55)
	at org.sonar.scanner.phases.AbstractPhaseExecutor.execute(AbstractPhaseExecutor.java:94)
	at org.sonar.scanner.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:180)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:135)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:121)
	at org.sonar.scanner.scan.ProjectScanContainer.scan(ProjectScanContainer.java:288)
	at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:283)
	at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:261)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:135)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:121)
	at org.sonar.scanner.task.ScanTask.execute(ScanTask.java:48)
	at org.sonar.scanner.task.TaskContainer.doAfterStart(TaskContainer.java:84)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:135)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:121)
	at org.sonar.scanner.bootstrap.GlobalContainer.executeTask(GlobalContainer.java:121)
	at org.sonar.batch.bootstrapper.Batch.doExecuteTask(Batch.java:116)
	at org.sonar.batch.bootstrapper.Batch.executeTask(Batch.java:111)
	at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:63)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:60)
	at com.sun.proxy.$Proxy88.execute(Unknown Source)
	at org.sonarsource.scanner.api.EmbeddedScanner.doExecute(EmbeddedScanner.java:233)
	at org.sonarsource.scanner.api.EmbeddedScanner.runAnalysis(EmbeddedScanner.java:151)
	at org.sonarqube.gradle.SonarQubeTask.run(SonarQubeTask.java:99)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
	at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.doExecute(DefaultTaskClassInfoStore.java:141)
	at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:134)
	at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:121)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:731)
	at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:705)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:122)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:197)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:107)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:111)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:92)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:70)
	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:63)
	at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:88)
	at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:248)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:197)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:107)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:241)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:230)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:124)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:80)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:105)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:99)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:625)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:580)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:99)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.talanlabs.gitlab.api.v3.GitLabAPIException: {"message":"500 Internal Server Error"}
	at com.talanlabs.gitlab.api.v3.http.GitLabHTTPRequestor.handleAPIError(GitLabHTTPRequestor.java:378)
	at com.talanlabs.gitlab.api.v3.http.GitLabHTTPRequestor.to(GitLabHTTPRequestor.java:125)
	at com.talanlabs.gitlab.api.v3.http.GitLabHTTPRequestor.to(GitLabHTTPRequestor.java:95)
	at com.talanlabs.gitlab.api.v3.services.GitLabAPICommits.postCommitComments(GitLabAPICommits.java:135)
	at com.talanlabs.sonar.plugins.gitlab.GitLabApiV3Wrapper.addGlobalComment(GitLabApiV3Wrapper.java:278)
	... 79 common frames omitted
Caused by: java.io.IOException: Server returned HTTP response code: 500 for URL: https://git.faked.org/api/v3/projects/12/repository/commits/1b9fcb99d18685340f21db583ce4ac3eec0f0c0f/comments
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1894)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1492)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:263)
	at com.talanlabs.gitlab.api.v3.http.GitLabHTTPRequestor.parse(GitLabHTTPRequestor.java:336)
	at com.talanlabs.gitlab.api.v3.http.GitLabHTTPRequestor.to(GitLabHTTPRequestor.java:123)
	... 82 common frames omitted
```

**Comment by gabrie-allaigre** [comment]
Hi,
What is gitlab version ?
Thanks

**Comment by jangrewe** [comment]
I'm running `10.3.0`.

**Comment by gabrie-allaigre** [comment]
In sonarqube -> administration -> general -> gitlab, change version api by v4

**Comment by jangrewe** [comment]
Ah, i was about to say "That's already set to v4", but then i saw this was only in the "Authentication" section, not in the "Reporting" section (where it says "Default: v3", btw).

Unfortunately it still throws that error...

Another thing: i'm passing ` -Dsonar.analysis.mode=preview` on the command line, and also set `property "sonar.analysis.mode", "preview"` in the Gradle file, but in the log i get `The use of the issues mode (sonar.analysis.mode=issues) is deprecated. This mode will be dropped in the future.` nonetheless.

**Comment by gabrie-allaigre** [comment]
What is version sonar-gitlab-plugin ?
Set GitLab API version is page down.
Plugin work with preview and issues

**Comment by jangrewe** [comment]
Yes, like i said, i've set the API version for both plugins to `v4`.
I have `GitLab Plugin for Reporting 2.1.0`.

**Comment by jangrewe** [comment]
Judging by the error message `Incorrect string value: '\xF0\x9F\x9A\xAB</...'` this seems to be a mismatch of the `notes` table's `note_html` column and the string we're trying to insert.
The column in my database is of type `mediumtext` with the collation `utf8mb4_general_ci`, does that match with your setup? 
The maximum length for `mediumtext` is 16,777,215 character, so length shouldn't be an issue, and the collation also looks like a solid choice.

**Comment by gabrie-allaigre** [comment]
sorry, I had misread your first message.
Do you use your template for global comment ?

**Comment by jangrewe** [comment]
I haven't configured any custom templates so far, this is a fresh sonarqube instance with only minimal configuration.

**Comment by jangrewe** [comment]
If i set `sonar.gitlab.disable_global_comment` to true, the error does not show up.

**Comment by gabrie-allaigre** [comment]
did you change the installation of gitlab?
I use postgres, maybe you have to ask on the issue tracker of gitlab

**Comment by jangrewe** [comment]
No really, it's an installation from source with MySQL, all according to their docs. But yeah, maybe it's an Postgres vs. MySQL problem...
Is there maybe a way to just disable the emojis, as they seem to be the cause for the MySQL error?

**Comment by jangrewe** [comment]
scratch that, i just saw the "Template Default with Images" ;-)

**Comment by jangrewe** [comment]
Alright, so with the alternative templates and the global comment enabled again it passes, BUT the "external" job still fails and has this tooltip:
```
sonarqube - SonarQube reported 6 issues, with 1 critical (fail) and 3 major and 1 minor and 1 info
```

**Comment by jangrewe** [comment]
Oh, i guess it doesn't really fail, that just the build-breaker... the ciritical issue is what makes the build fail!
Alright then, i think i got everything running again.

Thanks for being my rubber ducky! ;-)

**Comment by gabrie-allaigre** [comment]
Good Thanks

**Comment by kamihouse** [comment]
And I get this:

```
[GITLAB_bigdata-backend_branches] $ /opt/sonar/sonar-scanner-3.0.3.778-linux/bin/sonar-scanner -Dsonar.host.url=https://cd....******** -Dsonar.gitlab.project_id=4 -Dsonar.gitlab.ref_name=${CENV_GITLAB_BRANCH} -Dsonar.analysis.mode=preview -Dsonar.issuesReport.console.enable=true -Dsonar.gitlab.commit_sha=0cc0d0375a5a136b099fcd46599cd6a5cca9a156 -Dsonar.projectBaseDir=/var/lib/jenkins/workspace/GITLAB_bigdata-backend_branches
INFO: Scanner configuration file: /opt/sonar/sonar-scanner-3.0.3.778-linux/conf/sonar-scanner.properties
INFO: Project root configuration file: /var/lib/jenkins/workspace/GITLAB_bigdata-backend_branches/sonar-project.properties
INFO: SonarQube Scanner 3.0.3.778
INFO: Java 1.8.0_121 Oracle Corporation (64-bit)
INFO: Linux 3.13.0-92-generic amd64
INFO: User cache: /var/lib/jenkins/.sonar/cache
INFO: Issues mode
WARN: The use of the issues mode (sonar.analysis.mode=issues) is deprecated. This mode will be dropped in the future.
INFO: Load global settings
INFO: Load global settings (done) | time=818ms
INFO: Server id: AV0ZO9m-CthEaKUi4vL6
INFO: User cache: /var/lib/jenkins/.sonar/cache
INFO: Exclude plugins: devcockpit, ldap, authgithub, authbitbucket, pdfreport, authaad, googleanalytics, governance
INFO: Load plugins index
INFO: Load plugins index (done) | time=200ms
INFO: Load/download plugins
INFO: Load/download plugins (done) | time=120ms
INFO: SonarQube server 7.0.0
INFO: Default locale: "en_US", source code encoding: "UTF-8"
INFO: Process project properties
INFO: Load project repositories
INFO: Load project repositories (done) | time=310ms
INFO: Scanning only changed files
INFO: Execute project builders
ERROR: Unable to update commit status
com.talanlabs.gitlab.api.v4.GitLabAPIException: {"message":"500 Internal Server Error"}
	at com.talanlabs.gitlab.api.v4.http.GitLabHTTPRequestor.handleAPIError(GitLabHTTPRequestor.java:378)
	at com.talanlabs.gitlab.api.v4.http.GitLabHTTPRequestor.to(GitLabHTTPRequestor.java:125)
	at com.talanlabs.gitlab.api.v4.http.GitLabHTTPRequestor.to(GitLabHTTPRequestor.java:95)
	at com.talanlabs.gitlab.api.v4.services.GitLabAPICommits.postCommitStatus(GitLabAPICommits.java:181)
	at com.talanlabs.sonar.plugins.gitlab.GitLabApiV4Wrapper.createOrUpdateSonarQubeStatus(GitLabApiV4Wrapper.java:211)
	at com.talanlabs.sonar.plugins.gitlab.CommitFacade.createOrUpdateSonarQubeStatus(CommitFacade.java:122)
	at com.talanlabs.sonar.plugins.gitlab.CommitProjectBuilder.build(CommitProjectBuilder.java:49)
	at org.sonar.scanner.scan.ProjectBuildersExecutor.execute(ProjectBuildersExecutor.java:52)
	at org.sonar.scanner.scan.InputModuleHierarchyProvider.provide(InputModuleHierarchyProvider.java:39)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.picocontainer.injectors.MethodInjector.invokeMethod(MethodInjector.java:129)
	at org.picocontainer.injectors.MethodInjector.access$000(MethodInjector.java:39)
	at org.picocontainer.injectors.MethodInjector$2.run(MethodInjector.java:113)
	at org.picocontainer.injectors.AbstractInjector$ThreadLocalCyclicDependencyGuard.observe(AbstractInjector.java:270)
	at org.picocontainer.injectors.MethodInjector.decorateComponentInstance(MethodInjector.java:120)
	at org.picocontainer.injectors.CompositeInjector.decorateComponentInstance(CompositeInjector.java:58)
	at org.picocontainer.injectors.Reinjector.reinject(Reinjector.java:142)
	at org.picocontainer.injectors.ProviderAdapter.getComponentInstance(ProviderAdapter.java:96)
	at org.picocontainer.DefaultPicoContainer.getInstance(DefaultPicoContainer.java:699)
	at org.picocontainer.DefaultPicoContainer.getComponent(DefaultPicoContainer.java:647)
	at org.sonar.core.platform.ComponentContainer$ExtendedDefaultPicoContainer.getComponent(ComponentContainer.java:62)
	at org.picocontainer.DefaultPicoContainer.getComponent(DefaultPicoContainer.java:632)
	at org.picocontainer.parameters.BasicComponentParameter$1.resolveInstance(BasicComponentParameter.java:118)
	at org.picocontainer.parameters.ComponentParameter$1.resolveInstance(ComponentParameter.java:136)
	at org.picocontainer.injectors.SingleMemberInjector.getParameter(SingleMemberInjector.java:78)
	at org.picocontainer.injectors.ConstructorInjector$CtorAndAdapters.getParameterArguments(ConstructorInjector.java:309)
	at org.picocontainer.injectors.ConstructorInjector$1.run(ConstructorInjector.java:335)
	at org.picocontainer.injectors.AbstractInjector$ThreadLocalCyclicDependencyGuard.observe(AbstractInjector.java:270)
	at org.picocontainer.injectors.ConstructorInjector.getComponentInstance(ConstructorInjector.java:364)
	at org.picocontainer.injectors.AbstractInjectionFactory$LifecycleAdapter.getComponentInstance(AbstractInjectionFactory.java:56)
	at org.picocontainer.behaviors.AbstractBehavior.getComponentInstance(AbstractBehavior.java:64)
	at org.picocontainer.behaviors.Stored.getComponentInstance(Stored.java:91)
	at org.picocontainer.DefaultPicoContainer.getInstance(DefaultPicoContainer.java:699)
	at org.picocontainer.DefaultPicoContainer.getComponent(DefaultPicoContainer.java:647)
	at org.sonar.core.platform.ComponentContainer$ExtendedDefaultPicoContainer.getComponent(ComponentContainer.java:62)
	at org.picocontainer.DefaultPicoContainer.getComponent(DefaultPicoContainer.java:678)
	at org.sonar.core.platform.ComponentContainer.getComponentByType(ComponentContainer.java:265)
	at org.sonar.scanner.scan.ProjectScanContainer.doBeforeStart(ProjectScanContainer.java:118)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:133)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:121)
	at org.sonar.scanner.task.ScanTask.execute(ScanTask.java:48)
	at org.sonar.scanner.task.TaskContainer.doAfterStart(TaskContainer.java:84)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:135)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:121)
	at org.sonar.scanner.bootstrap.GlobalContainer.executeTask(GlobalContainer.java:121)
	at org.sonar.batch.bootstrapper.Batch.doExecuteTask(Batch.java:116)
	at org.sonar.batch.bootstrapper.Batch.executeTask(Batch.java:111)
	at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:63)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:60)
	at com.sun.proxy.$Proxy0.execute(Unknown Source)
	at org.sonarsource.scanner.api.EmbeddedScanner.doExecute(EmbeddedScanner.java:233)
	at org.sonarsource.scanner.api.EmbeddedScanner.runAnalysis(EmbeddedScanner.java:151)
	at org.sonarsource.scanner.cli.Main.runAnalysis(Main.java:123)
	at org.sonarsource.scanner.cli.Main.execute(Main.java:77)
	at org.sonarsource.scanner.cli.Main.main(Main.java:61)
Caused by: java.io.IOException: Server returned HTTP response code: 500 for URL: https://gitlab..../api/v4/projects/4/statuses/0cc0d0375a5a136b099fcd46599cd6a5cca9a156
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1876)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1474)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:254)
	at com.talanlabs.gitlab.api.v4.http.GitLabHTTPRequestor.parse(GitLabHTTPRequestor.java:336)
	at com.talanlabs.gitlab.api.v4.http.GitLabHTTPRequestor.to(GitLabHTTPRequestor.java:123)
	... 60 common frames omitted
```
